packages/rocketchat-lib/server/functions/Notifications.js

```diff
@@ -134,10 +139,11 @@ RocketChat.Notifications.streamRoom.allowWrite(function(eventName, username, typ
 
 		const user = Meteor.users.findOne(this.userId, {
 			fields: {
-				username: 1
+				username: 1,
+				name: 1
 			}
 		});
-		if (user != null && user.username === username) {
+		if (user != null && (user.username === username || (useRealNames() && (user.name === username)))) {
```

Comment:
Why did you create a function called `userRealNames` if you use only one time? Why no put `RocketChat.settings.get('UI_Use_Real_Name')` directly on this if instead?